### PR TITLE
PER 846: Added ARM Benchmark on AWS Graviton EC2

### PR DESCRIPTION
**Pull request by alfer-edb**
Added benchmark for AWS EC2 Graviton

### CODE REVIEW

File: aws-dbt2-ec2-arm/02_deploy/playbook-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  collections:
+    - edb_devops.edb_postgres
+
+  pre_tasks:
+    - name: Create {{ pg_owner }} OS user
+      ansible.builtin.user:
+        name: "{{ pg_owner }}"
+
+  roles:
+    - role: setup_dbt2_driver
```
Comment (hannahms) [inline aws-dbt2-ec2-arm/02_deploy/playbook-deploy.yml:49]: iirc, the `setup_repo` role needs to be run on the driver for the epel repo - Mark can confirm this

Comment (alfer-edb) [reply aws-dbt2-ec2-arm/02_deploy/playbook-deploy.yml:49]: I don't see a dependency from EPEL looking at s`etup_dbt2/validate_setup_dbt2.yml `and `setup_dbt2/dbt2_install_packages_driver.yml.` @mw2q  can you confirm?

File: aws-dbt2-ec2-arm/00_validate/run.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+ANSIBLE_ROLES_PATH=$ANSIBLE_ROLES_PATH \
+    ansible-playbook "${SOURCEDIR}/run.yml" \
+        -e "overrides=${OVERRIDES}" \
+        -e "root_directory=${ROOT_DIRECTORY}" \
+        -e "benchmark_directory=${BENCHMARK_DIRECTORY}" \
+        -e "results_directory=${RESULTS_DIRECTORY}" \
+        -e "terraform_project_name=${TERRAFORM_PROJECT_NAME}" \
+        -e "terraform_project_path=${TERRAFORM_PROJECT_PATH}" \
+        -e "terraform_plan_filename=${TERRAFORM_PLAN_FILENAME}"
```
Comment (bryan-bar) [inline aws-dbt2-ec2-arm/00_validate/run.sh:22]:
```suggestion
```suggestion
SOURCEDIR="$(realpath "$(dirname "${BASH_SOURCE[0]}")")"
BENCHMARK_DIRECTORY="$(realpath "${SOURCEDIR}/..")"
ROOT_DIRECTORY="$(realpath "${BENCHMARK_DIRECTORY}/..")"
ANSIBLE_ROLES_PATH="${ROOT_DIRECTORY}/roles"

ANSIBLE_ROLES_PATH=$ANSIBLE_ROLES_PATH \
    ansible-playbook "${SOURCEDIR}/run.yml" \
        -e "root_directory=${ROOT_DIRECTORY}" \
        -e "benchmark_directory=${BENCHMARK_DIRECTORY}"
```
```

Comment (bryan-bar) [inline aws-dbt2-ec2-arm/00_validate/run.sh:22]: The boilerplate changed a bit. The `benchmark-step` role's validate task will fill out the terraform and results variables based on the root directory and the benchmark directory.
```suggestion
SOURCEDIR="$(realpath "$(dirname "${BASH_SOURCE[0]}")")"
BENCHMARK_DIRECTORY="$(realpath "${SOURCEDIR}/..")"
ROOT_DIRECTORY="$(realpath "${BENCHMARK_DIRECTORY}/..")"
ANSIBLE_ROLES_PATH="${ROOT_DIRECTORY}/roles"

ANSIBLE_ROLES_PATH=$ANSIBLE_ROLES_PATH \
    ansible-playbook "${SOURCEDIR}/run.yml" \
        -e "root_directory=${ROOT_DIRECTORY}" \
        -e "benchmark_directory=${BENCHMARK_DIRECTORY}"
```

**Review by bryan-bar — CHANGES_REQUESTED** [review]

**Comment by alfer-edb** [comment]
Updated `aws-dbt2-ec2-arm/00_validate/run.sh` and added repos for dbt2-driver

**Comment by alfer-edb** [comment]
Updated edb-ansible https://github.com/EnterpriseDB/edb-ansible/pull/599

File: aws-dbt2-ec2-arm/04_execute/playbook-dbt2-run.yml [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    - name: Calculate drivers per processor
+      ansible.builtin.set_fact:
+        fpp: "{{ ((dbt2_connections | int) / ansible_processor_nproc) | int  }}"
+
+    # Convert the ramp up time minutes to milliseconds and calculate connection
+    # delay based on the targeted ramp up time.
+    - name: Calculate connection delay
+      ansible.builtin.set_fact:
+        connection_delay: "{{ ((dbt2_rampup | int) * 60000 / ((fpp | int) * \
+                          ansible_processor_nproc)) | int }}"
```
Comment (bryan-bar) [inline aws-dbt2-ec2-arm/04_execute/playbook-dbt2-run.yml:41]: `fpp` equals `0` when `dbt2_connections` is set to a low enough value and rounds down with the `int` filter. This is present in all dbt2 benchmarks and can be fixed in a future patch.

**Review by bryan-bar — APPROVED** [review]
LGTM

**Pull request merged**